logic/src/commonMain/kotlin/com/wire/kalium/logic/data/mlspublickeys/MLSPublicKeysMapper.kt

```diff
@@ -19,25 +19,15 @@
 package com.wire.kalium.logic.data.mlspublickeys
 
 import com.wire.kalium.cryptography.ExternalSenderKey
-import com.wire.kalium.network.api.base.authenticated.serverpublickey.MLSPublicKeysDTO
-import io.ktor.util.decodeBase64Bytes
+import com.wire.kalium.logic.data.mls.CipherSuite
 
 interface MLSPublicKeysMapper {
-    fun fromDTO(publicKeys: MLSPublicKeysDTO): List<MLSPublicKey>
     fun toCrypto(publicKey: MLSPublicKey): com.wire.kalium.cryptography.Ed22519Key
     fun toCrypto(externalSenderKey: ExternalSenderKey): com.wire.kalium.cryptography.Ed22519Key
+    fun fromCipherSuite(cipherSuite: CipherSuite): MLSPublicKeyType
 }
 
 class MLSPublicKeysMapperImpl : MLSPublicKeysMapper {
-    override fun fromDTO(publicKeys: MLSPublicKeysDTO) = with(publicKeys) {
-        removal?.entries?.mapNotNull {
-            when (it.key) {
-                ED25519 -> MLSPublicKey(Ed25519Key(it.value.decodeBase64Bytes()), KeyType.REMOVAL)
```

Comment:
this mapper will always return a list containing one item or null, so it was removed with choosing the correct signature from the map